Specific usage of method references fails to compile

#### Issue details
With a specific type of method reference used, an ArrayIndexOutOfBoundsException is thrown.

A method reference to an instance method of a given class, used from within that class.

#### Reproduction steps/code
The follow code will fail to compile:
```java
public class BrokenFile {

	public interface MyInterface {

		void invoke (BrokenFile brokenFile);

	}

	public void main () {
		test(BrokenFile::methodReferenceTarget);
	}

	public void methodReferenceTarget () {}

	private void test (MyInterface myInterface) {
		myInterface.invoke(this);
	}
}
```

#### Configuration
**Build Tools:**
- [X] IDEA plugin
- [X] Eclipse plugin
- [X] Gradle plugin
 
**Versions:**
_Please provide the version of RoboVM, XCode and JDK used_
- Robovm: 2.3.2-SNAPSHOT
- XCode: 8.3.3
- JDK: 1.8.0_73

**Build Targets:**
All targets

---

#### Stacktrace
```java
[ERROR] Couldn't compile app
org.robovm.compiler.CompilerException: java.lang.ArrayIndexOutOfBoundsException: -1
    at org.robovm.compiler.plugin.lambda.LambdaPlugin.transformMethod(LambdaPlugin.java:242)
    at org.robovm.compiler.plugin.lambda.LambdaPlugin.beforeClass(LambdaPlugin.java:89)
    at org.robovm.compiler.ClassCompiler.compile(ClassCompiler.java:681)
    at org.robovm.compiler.ClassCompiler.compile(ClassCompiler.java:292)
    at org.robovm.compiler.AppCompiler.compile(AppCompiler.java:309)
    at org.robovm.compiler.AppCompiler.compile(AppCompiler.java:392)
    at org.robovm.compiler.AppCompiler.compile(AppCompiler.java:452)
    at org.robovm.compiler.AppCompiler.build(AppCompiler.java:775)
    at org.robovm.idea.compilation.RoboVmCompileTask$3.doCompile(RoboVmCompileTask.java:228)
    at org.robovm.idea.compilation.RoboVmCompilerThread.run(RoboVmCompilerThread.java:66)
Caused by: java.lang.ArrayIndexOutOfBoundsException: -1
    at java.util.ArrayList.elementData(ArrayList.java:418)
    at java.util.ArrayList.get(ArrayList.java:431)
    at java.util.Collections$UnmodifiableList.get(Collections.java:1309)
    at org.robovm.compiler.plugin.lambda.LambdaClassGenerator.pushArguments(LambdaClassGenerator.java:203)
    at org.robovm.compiler.plugin.lambda.LambdaClassGenerator.createForwardingMethod(LambdaClassGenerator.java:134)
    at org.robovm.compiler.plugin.lambda.LambdaClassGenerator.generate(LambdaClassGenerator.java:79)
    at org.robovm.compiler.plugin.lambda.LambdaPlugin.transformMethod(LambdaPlugin.java:180)
    ... 9 more
```

--- a/compiler/compiler/src/main/java/org/robovm/compiler/plugin/lambda/LambdaClassGenerator.java
+++ b/compiler/compiler/src/main/java/org/robovm/compiler/plugin/lambda/LambdaClassGenerator.java
@@ -182,7 +182,7 @@ public class LambdaClassGenerator {
         // invoke dynamic call. We need to handle that parameter separately as
         // it's not part of the method signature of the implementation
         boolean paramsContainReceiver = isInstanceMethod
-                & !caller.getName().equals(implMethod.getMethodRef().declaringClass().getName())
+//                && !caller.getName().equals(implMethod.getMethodRef().declaringClass().getName())
                 && parameters.size() > implMethod.getMethodRef().parameterTypes().size();
         int paramsIndex = 0;
         int localIndex = 1; // we start at slot index 1, because this occupies

